drivers/mspi/mspi_dw.h

```diff
@@ -10,23 +10,23 @@
  */
 
 /* CTRLR0 - Control Register 0 */
-#define CTRLR0_SPI_FRF_MASK	GENMASK(23, 22)
+#define CTRLR0_SPI_FRF_MASK	GENMASK(22, 21)
```

Comment:
> - **Change bit offsets for CTRLR0 register in driver**- I am not sure if these offsets differ between IP revisions, @anangl can you check this?

Apparently they differ. I based on the 1.03a databook, as that's the SSI revision used in the nRF EXMIF peripheral. I've checked also the 2.00a databook, and the offsets are the same there; they must have been changed later. The driver would need to support both sets then. I think it should be fine to select the proper one at build time, instead of using the SSIC_COMP_VERSION register for this.

Reply:
Thanks for checking this, I figured the offsets would differ. I have version 4.03a. I'd argue we should use two bindings to distinguish the difference here, similar to how linux does: https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/spi/snps%2Cdw-apb-ssi.yaml#L55

Maybe `snps_designware_ssi` and `snps_designware_ssi-v2`? We could also use `snps_designware_ssi-4.03a` based off the databook version

Reply:
Yes, a different compatible string seems to be a good solution. I think I'd rather go with `-v2`, as we don't know if this change was introduced exactly in version 4.03a.